src/Component.d.ts

```diff
@@ -13,18 +13,18 @@ export type ComponentSchema = {
   [propName: string]: ComponentSchemaProp<any>;
 };
 
-export class Component<P> {
+export class Component<C> {
   static schema: ComponentSchema;
   static isComponent: true;
-  constructor(props?: P | false);
+  constructor(props?: Partial<Omit<C, keyof Component<any>>> | false);
```

Comment:
If there was a way to do `Partial<Pick<C, keyof C.schema>>` that would be even more ideal since the component constructor looks at the component schema keys for props to set.